### PR TITLE
Allow value 0 in includeBlankOption - fix issue 1022

### PR DESCRIPTION
**Pull request by zonky2**
Allow value 0 in includeBlankOption - fix issue #1022

see https://github.com/contao/contao/issues/1022#issuecomment-575210705

### CODE REVIEW

**Comment by m-vo** [comment]
I think we should add some tests for this change to see if it supports all edge cases.

**Comment by m-vo** [comment]
Oh, and I didn't notice this before: You PR should target `contao/contao` not `contao/core-bundle`.

**Comment by zonky2** [comment]
shit - can I change it online? I did the PR directly on gitlab...

File: src/Resources/contao/library/Contao/Widget.php [hunk 1]
```diff
@@ -1181,7 +1181,7 @@ public static function optionSelected($strOption, $varValues)
 			return '';
 		}
 
-		return (\is_array($varValues) ? \in_array($strOption, $varValues) : $strOption == $varValues) ? ' selected' : '';
+		return \in_array($strOption, array_map('strval', (array) $varValues), true) ? ' checked' : '';
```
Comment (ausi) [inline src/Resources/contao/library/Contao/Widget.php:1184]: Why did you change `selected` to `checked` here?

Comment (zonky2) [reply src/Resources/contao/library/Contao/Widget.php:1184]: I copy your code...
https://github.com/contao/contao/issues/1022#issuecomment-575210705

Comment (ausi) [reply src/Resources/contao/library/Contao/Widget.php:1184]: I used your code from https://github.com/contao/contao/issues/1022#issuecomment-564702570 as the basis.
But making a pull request with just a copy of my code doesn’t bring us much further. What we need to do is described here: https://github.com/contao/contao/issues/1022#issuecomment-695349026

Comment (zonky2) [reply src/Resources/contao/library/Contao/Widget.php:1184]: hmm something is on the wrong rail here - I'll check it again

**Comment by ausi** [comment]
Closed in favor of #2350

**Comment by zonky2** [comment]
I don't really understand why the PR is closed - so far the behaviour is unchanged...

see https://github.com/menatwork/contao-multicolumnwizard-bundle/issues/63 - I would like to close that issue

@stefanheimes @ausi

**Comment by ausi** [comment]
> I don't really understand why the PR is closed

It was closed in favor of contao/contao#2350

**Comment by zonky2** [comment]
entweder stehe ich auf dem Schlauch oder verstehe es nicht - warum ist #2350 geschlossen?!?

**Comment by ausi** [comment]
> warum ist #2350 geschlossen?!?

See https://github.com/contao/contao/pull/2350#issuecomment-735695944 :

> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs.

You can answer the open questions and address the comments to get it reopened.
